src/coreclr/jit/fgbasic.cpp

```diff
@@ -6734,6 +6734,10 @@ BasicBlock* Compiler::fgNewBBinRegionWorker(BBKinds     jumpKind,
 // Returns:
 //    The new block
 //
+// Notes:
+//    It is the responsibility of the caller to set the new block's handler index,
+//    if it is being inserted into a handler region, too.
+//
```

Comment:
Why do we not just always do the handler fixup?

Reply:
Without thinking about it too much, and without enough context... why are all these helpers better than the existing fgNewBBinRegion?